core/src/main/scala/cats/syntax/parallel.scala

```diff
@@ -108,9 +123,11 @@ trait ParallelFoldMapASyntax {
 }
 
 final class ParallelTraversableOps[T[_], A](private val ta: T[A]) extends AnyVal {
-  def parTraverse[M[_]: Monad, B](f: A => M[B])(implicit T: Traverse[T], P: Parallel[M]): M[T[B]] =
+  @deprecated("Kept for binary compatibility", "2.5.1")
+  def parTraverse[M[_], B](f: A => M[B], M: Monad[M], T: Traverse[T], P: Parallel[M]): M[T[B]] =
```

Comment:
shouldn't this have a second parameter list instead?

Reply:
WDYM? 🤔

Reply:
You moved the monad constraint in the first position of the implicit parameter list. My understanding of context bounds is that a new parameter list is appended to the method definition in this way:
```scala
   def parTraverse[M[_]: Monad, B](f: A => M[B])(implicit T: Traverse[T], P: Parallel[M]): M[T[B]] = 
   // is equivalent to
    def parTraverse[M[_], B](f: A => M[B])(implicit T: Traverse[T], P: Parallel[M])(implicit M: Monad[M]): M[T[B]] = 
 ```

Reply:
Nope, in Scala 2 you can have only one implicit list and the context bounds come first.

Reply:
Didn't know that, thanks for the explanation.

Reply:
But in the bytecode there is only one parameter list anyway (and no implicits) which is why both are binary compatible:
```scala
def parTraverse[M[_]: Monad, B](f: A => M[B])(implicit T: Traverse[T], P: Parallel[M]): M[T[B]]
def parTraverse[M[_], B](f: A => M[B], M: Monad[M], T: Traverse[T], P: Parallel[M]): M[T[B]]
```

That's a neat trick for library authors 😄